torch/csrc/jit/mobile/quantization.cpp

```diff
@@ -0,0 +1,64 @@
+#include <torch/csrc/jit/mobile/module.h>
+#include <torch/csrc/jit/mobile/quantization.h>
+
+namespace torch {
+namespace jit {
+namespace mobile {
+namespace quantization {
+
+void PTQQuanizationHelper::quantize_dynamic(
+    torch::jit::mobile::Module& m,
+    const std::string& method_name) {
+  std::string reset_observers_method_name = "reset_observers_" + method_name;
+  std::string observe_method_name = "observe_" + method_name;
+  std::string quantize_method_name = "quantize_" + method_name;
+  std::string quantized_method_name = "quantized_" + method_name;
+
+  TORCH_CHECK(
+      m.find_method(reset_observers_method_name).has_value(),
+      "PTQ ready module must have",
+      reset_observers_method_name,
+      " method.");
+  TORCH_CHECK(
+      m.find_method(observe_method_name),
+      "PTQ ready module must have",
+      reset_observers_method_name,
+      " method.");
+  TORCH_CHECK(
+      m.find_method(quantize_method_name),
+      "PTQ ready module must have",
+      quantize_method_name,
+      " method.");
+  TORCH_CHECK(
+      m.find_method(quantized_method_name),
+      "PTQ ready module must have",
+      quantized_method_name,
+      " method.");
+  TORCH_CHECK(
+      m.find_method("get_all_bundled_inputs"),
+      "PTQ ready module must have get_all_bundled_inputs method.");
+
+  auto inputs = m.run_method("get_all_bundled_inputs")
```

Comment:
Another dumb question: if the inputs are bundled aot, any reason we cannot replace the quantized method (i.e., call `PTQQuanizationHelper::quantize_dynamic`) aot? Which parts can only be observed at runtime?

Reply:
No we cannot do aot because mode is not yet trained. Once the model is trained then only it is ready to be quanitzed. The training happens on the device and model does not leave the device.

Reply:
I see. I think I missed this context. Thanks!